python/pyspark/ml/util.py

```diff
@@ -63,7 +63,7 @@ def _randomUID(cls):
         Generate a unique unicode id for the object. The default implementation
         concatenates the class name, "_", and 12 random hex chars.
         """
-        return unicode(cls.__name__ + "_" + uuid.uuid4().hex[12:])
+        return unicode(cls.__name__ + "_" + uuid.uuid4().hex[:12])
```

Comment:
`uuid.uuid4().hex[-12:]`? In Scala side, it is `UUID.randomUUID().toString.takeRight(12)`.

Reply:
Yea, `takeRight(12)` should be equivalent with `hex[-12:]`.